### DISCUSSION TITLE
Verify option { field: ["*"] }

### DISCUSSION THREAD

**Discussion by dungkaka** [mikro-orm/mikro-orm#5139] [General]
I have entities City and School, that are OneToMany relationship.
```Typescript
@Entity()
class City  {
    @OneToMany(() => School, (s) => s.city, { ref: true })
    schools = new Collection<School>(this, undefined, false);
}

@Entity()
class School {
    @ManyToOne(() => City, { ref: true })
    city: Ref<City>;
}


const city = await this.em.find(City, { id: 1 }, { fields: ["*"] });
// Seem city will populate all relationships when fields: ["*"]
// Expected result: city without populate anything
// Real result: city will populate all relationships
```

It causes maybe due to new feature of populate: ["*"] on v6. I just want to verify it then I can refactor my code. Thank you so much!

MikroORM: 6.0.4
NodeJs: 20

**Comment by B4nan** [comment]
Why do you set `fields: ['*']` in the first place? What you want is the default behavior, just drop that (or provide `fields: undefined` instead).

**Reply by dungkaka** [reply]
I use it v5, for example, in case when I want to just select a field from school:
```Typescript
const city = await this.em.find(City, { id: 1 }, { populate: ["schools"], fields: ["*", "schools.name"] });
```
Assume that my entity city have many relationship, and I have a query that populate schools and only some fields of school, example "name". Because field: " * " now will populate all things, I can not using  " * "  anymore, so now I need explicit declare all field of city in fields:
 
```Typescript
const city = await this.em.find(City, { id: 1 }, { populate: ["schools"], fields: ["id", "name", "createdAt", "updateAt", "many other fields", "schools.name"] });
````
Yeah, if it change behavior in v6, I will refactor my code.

**Reply by B4nan** [reply]
Isn't this working the way you want already?

```ts
const city = await this.em.find(City, { id: 1 }, { populate: ["schools"], fields: ["schools.name"] });
```

Either way, I don't mind fixing this, should be pretty simple, just trying to find a workaround till it gets fixed.

**Reply by B4nan** [reply]
Hmm actually, the problem you are describing can only be present if you skip the `populate` hint, it won't be an issue in the code you provided in this thread, where you set `populate: ['school']`. There is no inference if you provide it.

**Reply by dungkaka** [reply]
Aha, you're right. It's my mistake in my example. Actually, I often use fields: ["*"] for retrieve lazy scalar property instead of using populate, what why I get many errors because of auto populate on my project when upgrading now. Sorry for my unclear example !

**Comment by dungkaka** [comment]
> Isn't this working the way you want already?
> 
> ```ts
> const city = await this.em.find(City, { id: 1 }, { populate: ["schools"], fields: ["schools.name"] });
> ```
> 
> Either way, I don't mind fixing this, should be pretty simple, just trying to find a workaround till it gets fixed.

Can you make this syntax also available with em.populate. For example:
```Typescript
const city = await this.em.find(City, { id: 1 } });
await this.em.populate(city, ["schools"], { fields: ["schools.name"])
// If I use this now, it will ignore all other fields of City, 
// I can solve it by { fields: ["*", "schools.name"] }
```
 
But I'm like this syntax instead of { fields: ["*", "schools.name"] } which I use before, because ["*"] will also get lazy property.

**Reply by B4nan** [reply]
It is already fixed in latest version.

**Reply by dungkaka** [reply]
Do you mean v6.0.5? Thank you for fixing many unwanted behaviors I meet on this release.
I've updated to v6.0.5. In this case, I'm not sure because  it's still need ["*", "schools.name"] instead of ["schools.name"].
If you're mentioning v6.0.5, I'll check more.

**Reply by B4nan** [reply]
Only if you started providing complete reproductions... It's hard to navigate in this stream of short and incomplete snippets.

**Reply by B4nan** [reply]
It sounds like you reported two problems (this and #5138) and now the last missing step is the combination of those.

**Reply by B4nan** [reply]
Here is a fix for the `em.populate` with partial loading, let's hope it won't break anything, it was quite tricky to find the way to handle this.

1c7b44652100aa4e940787b3ee7ca4e208039984
